version: '3.8'

services:
  db:
    image: mysql:8.0
    container_name: adventurexp_db
    environment:
      MYSQL_ROOT_PASSWORD: ${DB_PASSWORD}
      MYSQL_DATABASE: adventurexp_db
    ports:
      - "3306:3306"
    volumes:
      - db_data:/var/lib/mysql
      - ./src/main/resources/sql/init.sql:/docker-entrypoint-initdb.d/init.sql
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - adventurexp-network

  app:
    build: .
    container_name: adventurexp_app
    environment:
      SPRING_DATASOURCE_URL: jdbc:mysql://db:3306/adventurexp_db
      SPRING_DATASOURCE_USERNAME: ${DB_USERNAME}
      SPRING_DATASOURCE_PASSWORD: ${DB_PASSWORD}
    ports:
      - "8080:8080"
    depends_on:
      db:
        condition: service_healthy
    networks:
      - adventurexp-network

volumes:
  db_data:

networks:
  adventurexp-network:
    external: true
